src/Compilers/CSharp/Test/Semantic/Semantics/ReadOnlyStructsTests.cs

```diff
@@ -351,88 +351,128 @@ public struct S
         }
 
         [Fact]
-        public void ReadOnlyMethod_OverrideBaseMethod()
+        public void ReadOnlyMethod_Override_AssignThis()
         {
             var csharp = @"
-public struct S1
+public struct S
 {
-    // note: GetType can't be overridden
-    public override string ToString() => throw null;
-    public override int GetHashCode() => throw null;
-    public override bool Equals(object o) => throw null;
+    public int i;
 
-    readonly void M()
-    {
-        // should warn--non-readonly invocation
-        ToString();
-        GetHashCode();
-        Equals(null);
+    // error
+    public readonly override string ToString() => (i++).ToString();
+    public readonly override int GetHashCode() => (i++).GetHashCode();
+    public readonly override bool Equals(object o) => (i++).Equals(o);
```

Comment:
>(i++).Equals(o); [](start = 54, length = 16)

Your verification on `this` focuses on methods. Consider adding an accessor scenario too.

Reply:
It looks like this test already existed and passed before this PR. Which test covers the change to `ThisParameterSymbol.RefKind`?

---
In reply to: [270199041](https://github.com/dotnet/roslyn/pull/34468#discussion_r270199041) [](ancestors = 270199041)

Reply:
See `CodeGenReadonlyStructTests.ReadOnlyMembers_MetadataRoundTrip`

---
In reply to: [270200919](https://github.com/dotnet/roslyn/pull/34468#discussion_r270200919) [](ancestors = 270200919,270199041)